ksqldb-functional-tests/src/test/resources/historical_plans/json_sr_-_deserialize_anonymous_array_-_value/6.1.0_1597874026054/spec.json

```diff
@@ -120,24 +114,18 @@
           },
           "partitions" : 4,
           "valueSchema" : {
-            "type" : "object",
-            "properties" : {
-              "FOO" : {
-                "connect.index" : 0,
+            "oneOf" : [ {
+              "type" : "null"
+            }, {
+              "type" : "array",
+              "items" : {
                 "oneOf" : [ {
                   "type" : "null"
                 }, {
-                  "type" : "array",
-                  "items" : {
-                    "oneOf" : [ {
-                      "type" : "null"
-                    }, {
-                      "type" : "string"
-                    } ]
-                  }
+                  "type" : "string"
                 } ]
               }
-            }
```

Comment:
This is another incompatibility issue with QTT. 

The JSON_SR schema has an ARRAY field wrapped into another `FOO` object which should be `STRUCT<FOO ARRAY<STRING>>`. When creating a JSON-SR stream with `WRAP_SINGLE_VALUE=false`, the internal array is unwrapped and used in the stream as `FOO ARRAY<STRING>` only.  

However, JSON_SR with JsonSchemaConverter does not work like that. JSON_SR supports anonymous fields by registering a schema without a field name. This makes SR API to understands the record is anonymous and does not have a field. When attempting to deserialize an anonymous record using the SR schema with the wrapped array, then the deserialization fails because the schema is not the one used during serialization. However, before this PR, JSON_SR deserialization was done using the JSON mapper (no JsonSchemaConverter) which ignores the SR schema. This deserialization was using the KSQL schema to deserialize and everything looked good. Running INSERT VALUES fails, though, even before the PR 'cause serialization uses the JsonSchemaConverter.

I found the above incompatibility was only on QTT, though. If I run the tests in Production, when I serialize a value with wrapped fields, the JsonSchemaConverter adds the SR schema ID in the record for future reference when deserializing it. When deserializing it with JsonSchemaConverter, the API will look at the right schema ID and deserialize it using that schema, then pass it to KSQL which will unwrap the field. 

Seems the test case is compatible in Prod but not in QTT. This is another type behavior that QTT fails to validate corrrectly. I think we should fix QTT to use the Prod serializers instead of its own ones.